cmd/cost/get.go

```diff
@@ -0,0 +1,244 @@
+			accountSlice = append(accountSlice, accounts.Accounts[i].Id)
+		}
+
+		if accounts.NextToken == nil {
+			break
+		}
+
+		//Get accounts
+		accounts, err = org.ListAccountsForParent(&organizations.ListAccountsForParentInput{
+			ParentId:  OU.Id,
+			NextToken: accounts.NextToken,
+		})
+	}
+
+	return accountSlice
+}
+
+//Get the account IDs of all (not only immediate) accounts under OU
+func getAccountsRecursive(OU *organizations.OrganizationalUnit, org awsprovider.OrganizationsClient) []*string {
+	var accountsIDs []*string
+
+	//Populate OUs
+	OUs := getOUs(OU, org)
+
+	//Loop through all child OUs, get their costs, and store it to cost of current OU
+	for _, childOU := range OUs {
+		accountsIDs = append(accountsIDs, getAccountsRecursive(childOU, org)...)
+	}
+
+	//*accountsIDs = append(*accountsIDs, getAccounts(OU, org)...)
+	return append(accountsIDs, getAccounts(OU, org)...)
+}
+
+//Get immediate OUs (child nodes) directly under given OU
+func getOUs(OU *organizations.OrganizationalUnit, org awsprovider.OrganizationsClient) []*organizations.OrganizationalUnit {
+	//OUSlice stores OUs
+	var OUSlice []*organizations.OrganizationalUnit
+
+	//Get child OUs under parent OU
+	OUs, err := org.ListOrganizationalUnitsForParent(&organizations.ListOrganizationalUnitsForParentInput{
+		ParentId: OU.Id,
+	})
+
+	//Populate OUSlice with OUs by looping until OUs.NextToken is null
+	for {
+		if err != nil {
+			log.Fatalln("Unable to retrieve child OUs under OU:", err)
+		}
+
+		//Add OUs to slice
+		for childOU := 0; childOU < len(OUs.OrganizationalUnits); childOU++ {
+			OUSlice = append(OUSlice, OUs.OrganizationalUnits[childOU])
+		}
+
+		if OUs.NextToken == nil {
+			break
+		}
+
+		OUs, err = org.ListOrganizationalUnitsForParent(&organizations.ListOrganizationalUnitsForParentInput{
+			ParentId:  OU.Id,
+			NextToken: OUs.NextToken,
+		})
+	}
+
+	return OUSlice
+}
+
+//Get the account IDs of all (not only immediate) accounts under OU
+func getOUsRecursive(OU *organizations.OrganizationalUnit, org awsprovider.OrganizationsClient) []*organizations.OrganizationalUnit {
+	var OUs []*organizations.OrganizationalUnit
+
+	//Populate OUs by getting immediate OUs (direct nodes)
+	currentOUs := getOUs(OU, org)
+
+	//Loop through all child OUs. Append the child OU, then append the OUs of the child OU
+	for _, currentOU := range currentOUs {
+		OUs = append(OUs, currentOU)
+		OUs = append(OUs, getOUsRecursive(currentOU, org)...)
+	}
+
+	return OUs
+}
+
+//Get cost of given account
+func getAccountCost(accountID *string, ce awsprovider.CostExplorerClient, timePtr *string, cost *float64) {
+
+	start := strconv.Itoa(time.Now().Year()-1) + time.Now().Format("-01-") + "01" //Starting from the 1st of the current month last year i.e. if today is 2020-06-29, then start date is 2019-06-01
+	end := time.Now().Format("2006-01-02")
+	granularity := "MONTHLY"
+	metrics := []string{
+		"NetUnblendedCost",
+	}
+
+	switch *timePtr {
```

Comment:
explain why this is needed

Reply:
Why "NetUnblendedCost" or why metrics is needed? AWS API seems to require metrics